dds/DCPS/RTPS/TypeLookup.cpp

```diff
@@ -8,384 +8,654 @@
-  {
-    if (!(strm << ACE_OutputCDR::from_octet(stru.return_code))) {
+    bool must_understand = false;
+    if (!strm.read_parameter_id(member_id, field_size, must_understand)) {
       return false;
     }
-    switch (stru.return_code) {
-    case DDS::RETCODE_OK:
-      return strm << stru.result;
+
+    switch (member_id) {
+    case 206790757:
+      if (!(strm >> stru.type_ids)) {
+        return false;
+      }
+      break;
+    default:
+      if (must_understand) {
+        if (DCPS_debug_level >= 8) {
+          ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) unknown must_understand field(%u) in OpenDDS::XTypes::TypeLookup_getTypes_In\n"), member_id));
+        }
+        return false;
+      }
+      strm.skip(field_size);
+      break;
     }
+  }
+  return false;
+}
+
+
+void serialized_size(const Encoding& encoding, size_t& size,
+  const XTypes::TypeLookup_getTypes_Out& stru)
+{
+  size_t mutable_running_total = 0;
+  serialized_size_delimiter(encoding, size);
+
+  serialized_size_parameter_id(encoding, size, mutable_running_total);
+  serialized_size(encoding, size, stru.types);
+
+  serialized_size_parameter_id(encoding, size, mutable_running_total);
+  serialized_size(encoding, size, stru.complete_to_minimal);
+
+  serialized_size_list_end_parameter_id(encoding, size, mutable_running_total);
+}
+
+bool operator<<(Serializer& strm, const XTypes::TypeLookup_getTypes_Out& stru)
+{
+  const Encoding& encoding = strm.encoding();
+  size_t total_size = 0;
+  serialized_size(encoding, total_size, stru);
+  if (!strm.write_delimiter(total_size)) {
     return false;
   }
 
-  bool operator>>(Serializer& strm, XTypes::TypeLookup_getTypes_Result& stru)
-  {
-    CORBA::Long return_code;
-    if (!(strm >> return_code)) {
-      return false;
+  size_t size = 0;
+  serialized_size(encoding, size, stru.types);
+  if (!strm.write_parameter_id(41962193, size)) {
+    return false;
+  }
+  if (!(strm << stru.types)) {
+    return false;
+  }
+
+  size = 0;
+  serialized_size(encoding, size, stru.complete_to_minimal);
+  if (!strm.write_parameter_id(193881463, size)) {
+    return false;
+  }
+  return strm << stru.complete_to_minimal;
+}
+
+bool operator>>(Serializer& strm, XTypes::TypeLookup_getTypes_Out& stru)
+{
+  size_t total_size = 0;
+  if (!strm.read_delimiter(total_size)) {
+    return false;
+  }
+
+  const size_t start_pos = strm.pos();
+
+  unsigned member_id;
+  size_t field_size;
+  while (true) {
+
+    if (strm.pos() - start_pos >= total_size) {
+      return true;
     }
-    switch (return_code) {
-    case DDS::RETCODE_OK: {
-      OpenDDS::XTypes::TypeLookup_getTypes_Out tmp;
-      if (strm >> tmp) {
-        stru.result = tmp;
-        stru.return_code = return_code;
-        return true;
-      }
+
+    bool must_understand = false;
+    if (!strm.read_parameter_id(member_id, field_size, must_understand)) {
       return false;
     }
+
+    switch (member_id) {
+    case 41962193:
```

Comment:
Magic number and elsewhere.

Reply:
It's generated by `opendds_idl` that way.

Reply:
Yea, I didn't know where to put them. It seems like it'd be more awkward to do anything else.